src/targets/gpu/mlir.cpp

```diff
@@ -167,6 +168,51 @@ std::string mlir_print(F f, T x)
     return ss.str();
 }
 
+struct mlir_logger
+{
+    std::string diagnostics;
```

Comment:
I dont see this being used anywhere except to construct stringstream.

Reply:
We pull it out in the `THROW` I remember

Reply:
But `diagnostics` is never assigned a value. Its always an empty string.

Reply:
I didn't realize that stringstream doesn't work like the LLVM version where it takes the argument as a backing store.